tests/data/test457

```diff
@@ -20,7 +20,8 @@ aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
 30
 bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
 21;heresatest=moooo
-cccccccccccccccccccccccccccccccc
+cccccccccccccccccccccccccccccc
+c
```

Comment:
I'm curious: was this change needed to make the test work now? If so: why?

Reply:
Ah, indeed. We used to write excess body data, e.g. data exceeding `--max-filesize` and handle the error afterwards. With the PR, we write only the max amount and then error.

I had to do some trickery with the `\n`, so that the expected data ended with `\n`.